Indicate presence of undisplayed hidden files with empty message

jeremypw opened the pull request:
Fixes #1278 

BEFORE

<img width="1062" height="703" alt="Screenshot from 2025-11-03 09 28 38" src="https://github.com/user-attachments/assets/feb581ce-ac82-4229-82c1-2f462f7f1857" />

AFTER
 - Completely empty folder

<img width="750" height="557" alt="Screenshot from 2025-11-08 19 21 45" src="https://github.com/user-attachments/assets/46ab929e-15f6-4a3b-b7e2-0806778cec43" />

- Only hidden files

<img width="900" height="604" alt="Screenshot from 2026-02-09 17 56 12" src="https://github.com/user-attachments/assets/4da5be5f-fb28-480e-9c64-5c383a357505" />

- Hidden files temporarily revealed

<img width="886" height="595" alt="Screenshot from 2026-05-11 15 13 19" src="https://github.com/user-attachments/assets/01914ffc-7a52-4f68-ae24-b6b7a5fcbc98" />

jeremypw committed:
Add hidden_label, throttle updating

jeremypw committed:
Merge branch 'main' into jeremypw/fix-empty-label

teamcons commented:
if we mention hidden files we may as well straight up display a button to show them, instead of "lol go find it yourself" ?

jeremypw committed:
Merge branch 'main' into jeremypw/fix-empty-label

teamcons commented:
A one-time "display this folder" that lasts only till you change folder could be nice

jeremypw commented:
That would be possible. My only concern is that there would be no obvious way to revert the step without reloading or navigating away from the folder (the appmenu and context menu would still show hidden files are not shown). Its not a big deal though I guess.

jeremypw committed:
Add button to override show-hidden for folder

jeremypw committed:
Merge branch 'main' into jeremypw/fix-empty-label

jeremypw committed:
Tweak hidden files message

zeebok committed:
Merge branch 'main' into jeremypw/fix-empty-label

zeebok commented:
> I wonder whether the button should show below the text? Any styling required?

Using only mental images I feel like below might look cleaner

jeremypw committed:
Merge branch 'main' into jeremypw/fix-empty-label

jeremypw committed:
Adjust wording

jeremypw committed:
Show button below, centred

zeebok commented:
I think this is good to go if you fix the conflicts

jeremypw committed:
Merge branch 'main' into jeremypw/fix-empty-label

jeremypw commented:
Need to make compatible with recent changes to settings and actions.

zeebok committed:
Merge branch 'main' into jeremypw/fix-empty-label

jeremypw committed:
Merge branch 'main' into jeremypw/fix-empty-label

jeremypw committed:
Remove reference to non-existent action

jeremypw committed:
AbstractDirectoryView no longer updates app setting for "show-hidden"

jeremypw committed:
Allow revert show hidden override

jeremypw commented:
@zeebok Could you take another look when you have time? I have updated this for recent changes to `main` and also added a refinement allowing the hidden files temporarily shown using the override for that folder to be hidden again. This requires the overlay to continue to be shown but I do not think this is likely to be a problem if the user is only taking a peek at the files.

zeebok commented:
Yeah I can take a look tomorrow!

zeebok commented:
It all functions as you describe! Code looks good too. I do notice a strange delay of sorts with this new overlay. If I am in a folder with this message and go up to the parent, I see the folder change and the message overlay is still there for a moment. Also see this delay when pressing the Show/Hide button; slight moment after clicking for the button to update.

jeremypw committed:
Merge branch 'main' into jeremypw/fix-empty-label

jeremypw commented:
Thanks for the review.  I'll see if I can reduce the update delay before merging.

jeremypw committed:
Hide hidden label before change directory

jeremypw committed:
Change button label before reloading

jeremypw committed:
Remove unused code path
